src/main/java/org/jboss/tools/intellij/analytics/PreloadLanguageServer.java

```diff
@@ -14,10 +17,13 @@
   private final ICookie cookies = ServiceManager.getService(Settings.class);
 
   private void attachLanguageClient(final File cliFile) {
-    final String[] EXTENSIONS = {"xml", "json", "txt"};
+    List<String> supportedExtensions = new ArrayList(Platform.supportedManifestFiles); // To avoid changes in actual copy
+    supportedExtensions.removeIf(e -> e.equals("requirements-dev.txt")); // Because a file with txt extension is already present
```

Comment:
Still to much coupling with the original list as it will not work if a new duplicate is added.
You can rewrite as:
```
supportedExtensions.stream()
  .map(s -> s.substring(s.lastIndexOf('.') + 1)
  .distinct()
  .forEach(e -> xxx)
```